### PR TITLE
[docs] Add auto generated docs

### PR DESCRIPTION
**Pull request by phillbaker**
### Description
_Describe what this change achieves._

Adds auto generated docs for all resources

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._ 
Closes https://github.com/opensearch-project/terraform-provider-opensearch/issues/8

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by bbarani** [comment]
@prudhvigodithi @peterzhuamazon Can you please review and add your comments?

File: docs/index.md [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+The provider needs to be configured with an endpoint URL before it can be
+used.
+
+AWS Opensearch Service domains and OpenSearch clusters deployed on Kubernetes
+and other infrastructure are supported.
+
+Use the navigation to the left to read about the available resources.
+
+```terraform
+# Configure the Elasticsearch provider
```
Comment (prudhvigodithi) [inline docs/index.md:21]: Hey @phillbaker this shows as `Elasticsearch provider`, is this what generated by generated by https://github.com/hashicorp/terraform-plugin-docs ?, can you please check how is this `Elasticsearch` being inferred.
Thank you

File: docs/index.md [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+provider "elasticsearch" {
```
Comment (prudhvigodithi) [inline docs/index.md:22]: Also I see lot of references with elasticsearch.

**Review by prudhvigodithi — CHANGES_REQUESTED** [review]

File: docs/index.md [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+  url = "http://127.0.0.1:9200"
+}
+
+# Create an index template
+resource "elasticsearch_index_template" "template_1" {
```
Comment (prudhvigodithi) [inline docs/index.md:27]: should be as [opensearch_index_template](https://github.com/opensearch-project/terraform-provider-opensearch/blob/main/provider/provider.go#L217)

**Comment by phillbaker** [comment]
Updated, please take a look @prudhvigodithi

**Comment by prudhvigodithi** [comment]
Thanks @phillbaker just a thought should we add the supported version of OpenSearch in the readme so that users can starting using with the supported version?
Thank you

**Comment by prudhvigodithi** [comment]
Hey 

> Thanks @phillbaker just a thought should we add the supported version of OpenSearch in the readme so that users can starting using with the supported version?
> Thank you

@phillbaker I'm to start with this PR and update the readme later with the supported matrix versions.
Thank you

**Comment by prudhvigodithi** [comment]
One More change is I can see the reference to `kibana` as well, can you please check @phillbaker

**Comment by prudhvigodithi** [comment]
Hey @phillbaker can you please check my previous comments when you get a chance?
Thank you

**Comment by prudhvigodithi** [comment]
Hey @phillbaker just in case if you are not aware about the [OpenSearch public slack](https://opensearch.org/slack.html), please feel free to join for better reach and discuss more about the terraform repo :) .
Thank you

**Comment by phillbaker** [comment]
> should we add the supported version of OpenSearch in the readme

added

> I can see the reference to kibana

added

**Review by prudhvigodithi — APPROVED** [review]

**Comment by prudhvigodithi** [comment]
Thanks @phillbaker, @peterzhuamazon @bbarani can you please add your review?
Thank you

**Review by peterzhuamazon — APPROVED** [review]

**Pull request merged**